### PR TITLE
Improve useFetchAllFavoritedTracks fetch

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Uses new api that has a limit of 1000, so we dont need to loop. not sure if we need to do the data munging part with with the decodeHashId?

### CODE REVIEW

File: packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts [hunk 1]
```diff
@@ -18,36 +18,31 @@ const { getLocalSaves } = savedPageSelectors
...
-        offset
-      }
-    )
-    const { data: result } = await fetch(url).then((response) =>
-      response.json()
-    )
+  const url = apiClient.makeUrl(
+    `/users/${encodeHashId(currentUserId)}/favorites`,
+    {
+      user_id: currentUserId
```
Comment (raymondjacobson) [inline packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts:33]: I think we need to specify limit=10000 here

File: packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts [hunk 2]
```diff
@@ -18,36 +18,32 @@ const { getLocalSaves } = savedPageSelectors
...
-          trackId: decodeHashId(trackSave.favorite_item_id),
-          favoriteCreatedAt: trackSave.created_at
-        }))
-    )
-  }
-  return tracksAndTimestamps
+  return result
+    .filter((trackSave) => trackSave.favorite_type === 'SaveType.track')
+    .map((trackSave: APIFavorite) => ({
+      trackId: decodeHashId(trackSave.favorite_item_id) as number,
```
Comment (amendelsohn) [inline packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts:44]: yeah, I think we still need this

File: packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts [hunk 3]
```diff
@@ -18,36 +18,32 @@ const { getLocalSaves } = savedPageSelectors
...
+    created_at: string
+  }[]
+}
+
 export const fetchAllFavoritedTracks = async (currentUserId: number) => {
-  let tracksAndTimestamps: { trackId: ID; favoriteCreatedAt: string }[] = []
-  let loadMore = true
-  let offset = 0
-  // TODO: store results in state to avoid duplicate fetching
-  while (loadMore) {
```
Comment (amendelsohn) [inline packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts:26]: I assume we're okay with a hard cap of 10k downloaded favorites? We could keep the while loop but it won't come up much

**Review by sliptype — APPROVED** [review]

**Review by amendelsohn — APPROVED** [review]

File: packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/hooks/useFetchAllFavoritedTracks.ts:26]: yeah i think lets just cap it!

**Pull request merged**